src/server/utils/dates.py

```diff
@@ -67,3 +76,95 @@ def weeks_in_range(week_range: Tuple[int, int]) -> int:
         year = Year(y)
         acc += year.totalweeks()
     return acc + 1  # same week should lead to 1 week that will be queried
+
+def dates_to_ranges(values: Optional[Sequence[Union[Tuple[int, int], int]]]) -> Optional[Sequence[Union[Tuple[int, int], int]]]:
+    """
+    Converts a mixed list of dates and date ranges to an optimized list where dates are merged into ranges where possible.
+    e.g. [20200101, 20200102, (20200101, 20200104), 20200106] -> [(20200101, 20200104), 20200106]
+    (the first two values of the original list are merged into a single range)
+    """
+    if not values or len(values) <= 1:
+        return values
+
+    # determine whether the list is of days (YYYYMMDD) or weeks (YYYYWW) based on first element
+    try:
+        if (isinstance(values[0], tuple) and guess_time_value_is_day(values[0][0]))\
+            or (isinstance(values[0], int) and guess_time_value_is_day(values[0])):
+            return days_to_ranges(values)
+        elif (isinstance(values[0], tuple) and guess_time_value_is_week(values[0][0]))\
+            or (isinstance(values[0], int) and guess_time_value_is_week(values[0])):
+            return weeks_to_ranges(values)
+        else:
+            return values
+    except:
+        return values
+
+def days_to_ranges(values: Sequence[Union[Tuple[int, int], int]]) -> Sequence[Union[Tuple[int, int], int]]:
+    intervals = []
+
+    # populate list of intervals based on original values
+    for v in values:
+        if isinstance(v, int):
+            # 20200101 -> [20200101, 20200101]
+            intervals.append([time_value_to_date(v), time_value_to_date(v)])
+        else: # tuple
+            # (20200101, 20200102) -> [20200101, 20200102]
+            intervals.append([time_value_to_date(v[0]), time_value_to_date(v[1])])
+
+    intervals.sort(key=lambda x: x[0])
```

Comment:
you can get rid of that argument to `sort` and itll do a proper lexicographic ordering.  im pretty sure you should then be able set `merged[-1][1] = interval[1]` below instead of doing the max...  removing the max wont always work if you keep that sort key there though!

Reply:
Tried this, broke some tests. I think it's safe to keep the logic as-is (: